sycl/include/sycl/detail/pi.h

```diff
@@ -1186,6 +1196,14 @@ __SYCL_EXPORT pi_result piQueueFlush(pi_queue command_queue);
 __SYCL_EXPORT pi_result
 piextQueueGetNativeHandle(pi_queue queue, pi_native_handle *nativeHandle);
 
+/// Gets the native handle of a PI queue object.
+///
+/// \param queue is the PI queue to get the native handle of.
+/// \param nativeHandle is the native handle of queue or commandlist.
+/// \param IsImmCmdList indicates what the native handle is.
+__SYCL_EXPORT pi_result piextQueueGetNativeHandle2(
+    pi_queue queue, pi_native_handle *nativeHandle, bool *IsImmCmdList);
```

Comment:
We should generalize this and `piextQueueCreateWithNativeHandle2` API to not talk about immediate command-lists which are L0 specific. I suggest
```
int32_t nativeHandleDesc;
```
as int can convey more information if needed in the future.

@kbenzie: please take this change to UR

Reply:
Ok, done.

Reply:
> @kbenzie: please take this change to UR

@smaslov-intel done https://github.com/oneapi-src/unified-runtime/issues/392